File: 剑指offer/medium/58-II_左旋转字符串/58-II_左旋转字符串.go
func reverseLeftWords(s string, n int) string {
	runes := []rune(s)
	// 翻转前n个字符
	reverse(runes[:n])
	// 翻转剩余字符
	reverse(runes[n:])
	// 翻转整个字符串
	reverse(runes)
	return string(runes)
}

func reverse(arr []rune) {
	left, right := 0, len(arr)-1
	for left < right {
		arr[left], arr[right] = arr[right], arr[left]
		left++
		right--
	}
}

File: 剑指offer/medium/58-II_左旋转字符串/58-II_左旋转字符串.py
class Solution:
    def reverseLeftWords(self, s: str, n: int) -> str:
        s_list = list(s)
        # 翻转前n个字符
        s_list[:n] = reversed(s_list[:n])
        # 翻转剩余字符
        s_list[n:] = reversed(s_list[n:])
        # 翻转整个字符串
        s_list.reverse()
        return ''.join(s_list)